traits.yml

```diff
@@ -45,30 +58,123 @@ traits:
         properties:
           name:
             type: string
-            description: |
+            description: >
               The humanized name of entity itself, not including any
               hierarchy or classification.
 
+
               For example:
               - `"Cuttlefish v1"` - for a version of an asset
               - `"seq003"` - for a sequence in a hierarchy
 
           qualifiedName:
             type: string
-            description: |
-              An unambiguous, humanised name for the entity.
+            description: >
+              An unambiguous, humanized name for the entity.
+
 
               The display name may want to consider the host, and any other
               relevant Context information to form a display name for an
               entity that can uniquely identify the entity in that context.
 
+
               For example:
               - `"dive / build / cuttlefish / model / v1"` for a version
                   of an asset in an 'open recent' menu.
               - `"Sequence 003 [ Dive / Episode 1 ]"` for a sequence in
-                 an hierarchy as a window title.
+                 a hierarchy as a window title.
+  lifecycle:
+    description: >
+      Traits that are concerned with describing aspects of the lifecycle
+      of an entity, such as versioning.
+    members:
+      Version:
+        description: >
+          Describes a specific version of an entity.
+
+
+          We define a version as a revision or iteration of what is
+          otherwise the same logical entity. Versions are created when
+          new data is published to an existing entity reference of the
+          same trait set.
```

Comment:
Yup, we spent literally hours sketching that very situation :roll_eyes: 

Though is it worth a note that publishing to existing entities with a different trait set is also a thing, just not this thing?

Reply:
^ more apropos to https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation/pull/51#discussion_r1272164015 - will resolve this since it was mostly just my brain fail.